### PR TITLE
MOD-704 Decideid assets web framework

### PR DESCRIPTION
**Pull request by xwdeng**
---
### Summary
Add decideid assets web framework.

image
image
image

### Issues
*Link to the issue

### Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update

### Checklist
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published.

### Test
*How did you test your change?

### Additional Context
*Add any other context or screenshots about the pull request

### CODE REVIEW

File: src/decideid_assets/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  }, []);
+
+  return (
+    <>
+      <Header />
+      <ProfileProvider>
+        <Router history={history}>
+          <Switch>
+            <Route path="/app" component={AuthApp} />
+            <Route path="/" component={External} />
```
Comment (VitalyGitDev) [inline src/decideid_assets/src/App.tsx:39]: I'm not sure we need External routes at all

File: src/decideid_assets/src/components/app/AuthApp.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import { useConnect } from "@connect2icmodclub/react";
+
+
+export default function AuthApp() {
+  const history = useHistory();
+
+  const { isConnected, principal } = useConnect();
+
+  if (!isConnected)
```
Comment (VitalyGitDev) [inline src/decideid_assets/src/components/app/AuthApp.tsx:14]: This check must be only inside root-component

File: src/decideid_assets/src/contexts/profile.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    return (
+        <ProfileContext.Provider value={{ profile, updateProfile, createProfile, refreshProfile, clear }}>
+            {children}
+        </ProfileContext.Provider>
+    );
+};
+
+// Custom hook to use the profile context
+export const useProfile = (): ProfileContextType => {
```
Comment (VitalyGitDev) [inline src/decideid_assets/src/contexts/profile.tsx:88]: I believe that delivered UserProfile data must be as part of application state, cause its not an entity which can be changed very often.
Please add StateManagement as part of this application shell.

File: src/decideid_assets/package.json [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  "type": "module",
+  "keywords": [
+    "Internet Computer",
+    "Motoko",
+    "JavaScript",
+    "Canister"
+  ],
+  "scripts": {
+    "build": "node ../../scripts/build/gen_files_by_env.cjs && webpack",
+    "start": "export DEV_ENV=${DEV_ENV:-qa} &&  node ../../scripts/build/gen_files_by_env.cjs && webpack serve --mode development --config webpack.config.cjs"
```
Comment (VitalyGitDev) [inline src/decideid_assets/package.json:14]: you can simplify "start" like 
```
export DEV_ENV=${DEV_ENV:-qa} &&  npm run build && webpack serve --mode development --config webpack.config.cjs
```

File: src/decideid_assets/package.json [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    "mic-recorder-to-mp3": "^2.2.2",
+    "moment": "^2.29.4",
+    "postcss-preset-env": "^9.3.0",
+    "react": "^18.2.0",
+    "react-accessible-accordion": "^4.0.0",
+    "react-bulma-components": "^4.0.7",
+    "react-burger-menu": "^3.0.6",
+    "react-datepicker": "^4.8.0",
+    "react-dom": "^18.2.0",
+    "react-dropzone": "^11.4.2",
```
Comment (VitalyGitDev) [inline src/decideid_assets/package.json:91]: do we really will use dependencies like this?

Comment (xwdeng) [reply src/decideid_assets/package.json:91]: let me clean up more unused.

File: src/decideid_assets/package.json (hunk 4 above)
Comment (xwdeng) [reply src/decideid_assets/package.json:14]: I cannot run the build here. This is to run in debug mode. 
I will separate the gen_files_by_env call.

File: src/decideid_assets/src/contexts/profile.tsx (hunk 3 above)
Comment (xwdeng) [reply src/decideid_assets/src/contexts/profile.tsx:88]: Em. Could we delay this application state change ? I want to focus on the backend design for now.

File: src/decideid_assets/src/components/app/Main.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+
+export default function Main() {
+  const { principal } = useConnect();
+  const {profile, isLoading} = useProfile();
+
+  return (
+    <>
+      Your principal: {principal}
+      {!!!isLoading ? <p>Email: {profile?.email}</p> : null}
```
[the remark this answers is not in the document]
Comment (xwdeng) [reply src/decideid_assets/src/components/app/Main.tsx:16]: i lost my mind.. lol 
will clean up this in the next pr.

File: src/decideid_assets/src/hooks/useProfile.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  async function createProfileOnServer(newProfile: Profile): Promise<Profile> {
+    
+    throw new Error('createProfileOnServer function is not implemented.');
+  }
+
+  const { data: profile, error, isError, isLoading }: UseQueryResult<Profile, Error> = useQuery('profile', fetchProfile);
+
+  const updateProfileMutation: UseMutationResult<Profile, Error, Profile> = useMutation(updateProfileOnServer, {
```
[the remark this answers is not in the document]
Comment (xwdeng) [reply src/decideid_assets/src/hooks/useProfile.tsx:47]: got it. will consider this feature.

**Pull request merged**